packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -457,41 +495,13 @@ function measureBoxModelsMiddleware(): Middleware<Dispatch, State, Dispatch> {
 
 export function messageChannelMiddleware(
   client: MakeswiftHostApiClient,
+  channel: IMessageChannel,
 ): Middleware<Dispatch, State, Dispatch> {
-  return ({ dispatch, getState }: MiddlewareAPI<Dispatch, State>) =>
+  return ({ dispatch }: MiddlewareAPI<Dispatch, State>) =>
     (next: ReduxDispatch<Action>) => {
-      let cleanUp = () => {}
-
-      if (typeof window === 'undefined') return cleanUp
-
-      const messageChannel = new window.MessageChannel()
-
-      window.parent.postMessage(messageChannel.port2, '*', [messageChannel.port2])
-
-      messageChannel.port1.onmessage = (event: MessageEvent<Action>) => dispatch(event.data)
```

Comment:
Decoupled channel creation from the middleware creation, see `MessageChannel` below.